apps/kg/dataloader/sampler.py

```diff
@@ -29,35 +29,38 @@ def RelationPartition(edges, n):
         rel_cnts[idx] += 1
     for i, edge_cnt in enumerate(edge_cnts):
         print('part {} has {} edges and {} relations'.format(i, edge_cnt, rel_cnts[i]))
+
     parts = []
     for _ in range(n):
         parts.append([])
-    for h, r, t in edges:
-        idx = rel_dict[r]
-        parts[idx].append((h, r, t))
+    # let's store the edge index to each partition first.
+    for i, r in enumerate(rels):
+        part_idx = rel_dict[r]
+        parts[part_idx].append(i)
+    for i, part in enumerate(parts):
+        parts[i] = np.array(part, dtype=np.int64)
```

Comment:
Will parts[part_idx] overwritten by parts[i]?
It is doing in-place update.

Reply:
I'm not sure what you mean. `parts[part_idx]` isn't in this for loop.